src/agent/testresultreader.ts

```diff
@@ -263,20 +285,41 @@ export class ResultReader {
             var stackTrace = "";
             if (testCaseNode.failure) {
                 outcome = "Failed";
-                if (testCaseNode.failure.text) {
-                    stackTrace = testCaseNode.failure.text();
+                var testNode;
+
+                if (testCaseNode.failure.count() > 1) {
+                    this.command.warning("Multiple failures in the test case: " + testName + ". Picking the first for publishing!");
+                    testNode = testCaseNode.failure.at(0);
+                }
+                else {
+                    testNode = testCaseNode.failure;
+                }
+
+                if (testNode.text) {
+                    stackTrace = testNode.text();
                 }
-                if (testCaseNode.failure.attributes().message) {
-                    errorMessage = testCaseNode.failure.attributes().message;
+
+                if (testNode.attributes().message) {
+                    errorMessage = testNode.attributes().message;
                 }
             }
             else if (testCaseNode.error) {
                 outcome = "Failed";
-                if (testCaseNode.error.text) {
-                    stackTrace = testCaseNode.error.text();
+                var testNode;
+
+                if (testCaseNode.error.count() > 1) {
+                    this.command.warning("Multiple failures in the test case: " + testName + ". Picking the first for publishing!");
+                    testNode = testCaseNode.error.at(0);
+                }
+                else {
+                    testNode = testCaseNode.error;
+                }
+
+                if (testNode.text) {
+                    stackTrace = testNode.text();
                 }
-                if (testCaseNode.error.attributes().message) {
-                    errorMessage = testCaseNode.error.attributes().message;
+                if (testNode.attributes().message) {
+                    errorMessage = testNode.attributes().message;
                 }
             }
             else if (testCaseNode.skipped) {
```

Comment:
We cannot have a switch here. We are not checking the state of a single entity. The first if checks if there are failure nodes, the second should be executed only if there are no failures and is thus checking error nodes. Similarly third should be executed only when there are no failure and error nodes.